### ISSUE TITLE
Touchpad not working on monterey

### ISSUE THREAD

**Issue by thallesavelar22** [z3r0c00l-2k/Nitro-5-Hackintosh-OC#9]
After installing the macos, everything works, except the touchpad and trackpad. I still updated the kexts to a new one, but without success. The specs:

intel i5 9300h
24gb ram
UHD 630
Touchpad Synaptics

![image](https://user-images.githubusercontent.com/93916250/140730732-4c2bf051-ad5a-4f36-a4b6-262943974e38.png)

**Comment by z3r0c00l-2k** [comment]
Yeah. I found the issue too. I am using an external mouse and keyboard, so didn't notice earlier. Will investigate this coming weekend.

**Comment by thallesavelar22** [comment]
You can based on Velcis Ribeiro. I was using his EFI and was working without problems. Changed to you EFI because de power management and monterey.

Here's the link.
https://github.com/velcis/Nitro5-AN515-51

**Comment by thallesavelar22** [comment]
Any news?

**Comment by z3r0c00l-2k** [comment]
Sorry bro, got caught up in work, I have pushed a new branch called [touchpad-fix](https://github.com/z3r0c00l-2k/Nitro-5-Hackintosh-OC/tree/touchpad-fix). please check if that's working for you.

**Comment by magnouema** [comment]
the touchpad still doesn't work

**Comment by LuizRafael79** [comment]
Any success with this?

**Comment by MiddlemanTM** [comment]
Yes there is an solution! To get the trackpad to work you'll need to setup your Voodoo drivers like this:
![Screen Shot 2022-03-24 at 12 12 37 AM](https://user-images.githubusercontent.com/78093030/159755285-a802ccf7-7322-422c-b61a-9bb983238f72.png)

And some of these ACPI patches and SSDTs:

![Screen Shot 2022-03-24 at 1 04 38 AM](https://user-images.githubusercontent.com/78093030/159755764-89fba1d5-92b0-4780-bc6a-aa5b06ae2a8f.png)
![Screen Shot 2022-03-24 at 1 04 46 AM](https://user-images.githubusercontent.com/78093030/159755775-4e0792e0-4830-4f5a-8c75-19a31e786e6e.png)

**Comment by kalilmaciel** [comment]
I had some troubles with the trackpad with my device. I fix it taking some precautions:
1 - If you want to use the trackpad, disconect any mouse before turn on your device.
2 - This model of notebook has a special shortcut to disable the trackpad. If your trackpad is not working, try press Fn + F7 and restart your notebook (with no mouse connected). This should work.

I hope helped you.

**Comment by AlBannaTechno** [comment]
@MiddlemanTM 
can you please share your EFI, i tried so many things to make trackpad works,
but nothing works

thanks.

**Comment by MiddlemanTM** [comment]
@AlBannaTechno,

Sorry it has taken this long to get back to you - didn't realise you had a query.
My EFI can be found here > https://www.tonymacx86.com/threads/success-hp-notebook-17-x105tx-intel-core-i7-7500u-amd-r7-m260-10-13-6-guide.309923/

As far as I know aside from using the Voodoo drivers, the order of the kext and plists listed are just as vital. I had to disable two of the kexts as shown above in order for the system to recognise (the system cannot accept more than one load-up of the same driver). Also I did not load any form of .efi USB drivers at all at boot, just the three standard OpenHFSPlus, OpenRuntime and OpenCanopy.efis.

**Comment by Reyhankeselek** [comment]
I tried appleps2smarttouchpad , voodooi2c, voodooRMI and vodoops2touchpad and it still did not work here my efi :  https://github.com/Reyhankeselek/Lenovo-ideapad-slim-3-15IML05-hackintosh
(select the latest release)

**Comment by MiddlemanTM** [comment]
Okay guys, for working Synaptics trackpad you now only need to use the following setup.
This is now the official fix according to Acidanthera:

VoodooPS2controller + VoodooRMI + VoodooSMBus

> https://github.com/acidanthera/bugtracker/issues/1579
